src/homework14/Box.java

```diff
@@ -0,0 +1,64 @@
+package homework14;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class Box <T>{
+    public List<Fruit> boxList = new ArrayList<>();
```

Comment:
`List<Fruit> boxList` - this means that the `boxList` variable can hold any type of `Fruit` i.e. `Apple` and `Orange`.  This is incorrect - the `Box` (and object holder property - `boxList` under the hood) can hold only `Apple` or `Orange`.
In terms of generics, you have to use `T` as a list type, thus allowing you to create a `Box` with a specific type like `Apple` or `Orange`.